File: JAVA/lld-examples/easy/parking-lot-system/src/main/java/com/systemdesign/lld/parking/VehicleFactory.java
package com.systemdesign.lld.parking;

/**
 * Factory class for creating vehicles
 */
public class VehicleFactory {
    
    public static Vehicle createVehicle(VehicleType type, String licensePlate) {
        switch (type) {
            case CAR:
                return new Car(licensePlate);
            case MOTORCYCLE:
                return new Motorcycle(licensePlate);
            case TRUCK:
                return new Truck(licensePlate);
            default:
                throw new IllegalArgumentException("Unknown vehicle type: " + type);
        }
    }
} 

File: Python/lld-examples/easy/parking-lot-system/vehicle_factory.py
from vehicle import Vehicle, VehicleType, Car, Motorcycle, Truck

class VehicleFactory:
    """Factory class for creating vehicles"""
    
    @staticmethod
    def create_vehicle(vehicle_type: VehicleType, license_plate: str) -> Vehicle:
        """Create vehicle based on type"""
        if vehicle_type == VehicleType.CAR:
            return Car(license_plate)
        elif vehicle_type == VehicleType.MOTORCYCLE:
            return Motorcycle(license_plate)
        elif vehicle_type == VehicleType.TRUCK:
            return Truck(license_plate)
        else:
            raise ValueError(f"Unknown vehicle type: {vehicle_type}") 